app/models/shipit/stack.rb

```diff
@@ -528,8 +528,25 @@ def sync_github
       GithubSyncJob.perform_later(stack_id: id)
     end
 
+    alias links_orig links
+    def links
+      persisted_tasks = tasks.select(&:persisted?)
+      return links_orig if persisted_tasks.blank?
+
+      variables = Shipit::TaskCommands.new(persisted_tasks.last).env
```

Comment:
I have to think about that one. This seems quite a hack.

What variable do you need to interpolate here? 

I'm not against that interpolation, but if it could be done without loading the last deploy it would be preferable.

Reply:
Perhaps a practical example will be helpful. 

We run and deploy to a number of internal Kubernetes clusters. We have a custom task which chooses the _least active_ cluster and provisions a Kubernetes namespace for a stack when a new stack is added. We register internal hooks to deployments which capture information about the kubernetes cluster to which a stack is deployed and make that knowledge available via the deployment/task's `#env`. 

This information is useful to us because the cluster to which a stack is deployed determines the URL of its telemetry services - for example the pattern to get to a stack's logs looks something like `https://logs.${ENVIRONMENT}.application.${CLUSTER}.domain.com`. Does that help at least put a frame around the picture?

Reply:
> We register internal hooks to deployments which capture information about the kubernetes cluster to which a stack is deployed and make that knowledge available via the deployment/task's `#env`.

Ok, so you use tasks to persist data?

It's quite a clever hack, but from a product design standpoint, using the last stack for interpolation would make for a very surprising behavior.

What if:

  - We use `Stack#env` for interpolation (doesn't exist yet, but easy to add).
  - You can then do this cluster selection in your host app in ruby, and store it either in the db or in `Rails.cache`.

Reply:
> We use Stack#env for interpolation (doesn't exist yet, but easy to add).

Sounds like something I can chase down. Just want to make sure I'm on the same page. If this bit lives in `shipit-engine` what would your recommendation be to allow the host app - like mine - inject values into the stack's environment? Would it be appropriate to add another concept / association for these values to the Stack and use them to build the value of `Stack#env`? Perhaps setting one of these values in the host application looks something like `stack.envrionment_variable.create(name: "cluster", value: "cluster-1")`? Does this seem reasonable? Do you think the name `environment_variable` is appropriate here? If not, any suggestions?

Reply:
IMO the best way is:

```ruby
module ExtraStackEnv
  def env
    env = super
    # add whatever you want here
    env
  end
end

Shipit::Stack.prepend(ExtraStackEnv)
```

Reply:
Addressed by https://github.com/Shopify/shipit-engine/pull/1018/commits/bc0fd9d791729cfc0eb497c764f53df53112b38e and https://github.com/Shopify/shipit-engine/pull/1018/commits/a210081966d860400107e216f80446ab2f1f366a